consensus/consensus-types/src/proof_of_store.rs

```diff
@@ -211,6 +211,10 @@ impl SignedBatchInfoMsg {
     pub fn take(self) -> Vec<SignedBatchInfo> {
         self.signed_infos
     }
+
+    pub fn author(&self) -> PeerId {
+        self.signed_infos[0].signer()
```

Comment:
this directly panics if the vector is empty

Reply:
Thanks for pointing this out. Actually a similar syntax is used in BatchMsg too. Fixed that as well.